roxie/ccd/ccdstate.cpp

```diff
@@ -2588,12 +2588,30 @@ class CRoxiePackageSetManager : implements IRoxieQueryPackageManagerSet, impleme
             }
             else if (strieq(queryName, "control:memlock"))
             {
-                roxiemem::lockRoxieMem(true);
+                int srtn = roxiemem::lockRoxieMem(true);
+                if (!srtn)
+                {
+                    topology->setPropBool("@heapLockMemory", true);
+                    reply.append(" <heapLockMemory>1</heapLockMemory>\n");
+                }
+                else
+                    reply.appendf(" <Error=%d, unable to lock roxieMemory/>\n", srtn);
```

Comment:
This isn't valid xml - the receiver won't be able to parse it.

Maybe 
`<LockedMemory locked=%u [error=%s]/>
`?